eclipse-theia/theia
# Support customization of configuration area for customer application

#### What it does

Adds an application prop to define the location of the configuration area.
This property can be used by theia-based applications to customize their own configuration area. The logic of the THEIA_CONFIG_DIR environment variable should still works and it should override this application prop, without the need of overriding the environment variable server.

fixes #14318

Contributed on behalf of STMicroelectronics
#### How to test

- Sample application can be used without modification. '.theia' should be the default configuration folder (default prop is used)
- Setting an env variable to another folder (export THEIA_CONFIG_DIR=".theia-configDir") for example, and re-test with the sample application.
- Unset the environment variable and setting the application property in the sample application package.json to "configurationFolder" : ".theia-example". Config dir shall be $home/.theia-example
- Set again the env property. The environment property should override the configurationFolder value

#### Follow-ups

- Update theia-blueprint to use this application prop rather than then env variable server mechanism. (see https://github.com/eclipse-theia/theia-blueprint/issues/393)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
